File: payment-service/internal/model/payment.go
package model

import "time"

// Payment statuses.
const (
	StatusPending   = "pending"
	StatusProcessed = "processed"
	StatusRefunded  = "refunded"
	StatusFailed    = "failed"
)

// Payment represents a payment record stored in Redis.
type Payment struct {
	ID            string    `json:"id"`
	SagaID        string    `json:"saga_id"`
	OrderID       string    `json:"order_id"`
	CustomerID    string    `json:"customer_id"`
	Amount        float64   `json:"amount"`
	Status        string    `json:"status"`
	TransactionID string    `json:"transaction_id"`
	FailureReason string    `json:"failure_reason,omitempty"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

File: services/payment-service/app/Models/Payment.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Str;

class Payment extends Model
{
    protected $fillable = [
        'id',
        'tenant_id',
        'order_id',
        'customer_id',
        'amount',
        'currency',
        'status',
        'saga_id',
        'provider_reference',
        'refund_id',
        'metadata',
    ];

    protected $casts = [
        'amount'   => 'decimal:2',
        'metadata' => 'array',
    ];

    const STATUS_PENDING   = 'pending';
    const STATUS_PROCESSED = 'processed';
    const STATUS_FAILED    = 'failed';
    const STATUS_REFUNDED  = 'refunded';

    protected static function boot(): void
    {
        parent::boot();
        static::creating(function (Payment $payment) {
            if (empty($payment->id)) {
                $payment->id = (string) Str::uuid();
            }
        });
    }

    public function getKeyType(): string
    {
        return 'string';
    }

    public function getIncrementing(): bool
    {
        return false;
    }
}
